x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.ts

```diff
@@ -265,3 +257,32 @@ export const RelatedIntegration = t.exact(
  */
 export type RelatedIntegrationArray = t.TypeOf<typeof RelatedIntegrationArray>;
 export const RelatedIntegrationArray = t.array(RelatedIntegration);
+
+/**
+ * Schema for fields relating to investigation fields, these are user defined fields we use to highlight
+ * in various features in the UI such as alert details flyout and exceptions auto-population from alert.
+ * Added in INSERT PR
+ * Right now we only have a single field but anticipate adding more related fields to store various
+ * configuration states such as `override` - where a user might say if they want only these fields to
+ * display, or if they want these fields + the fields we select. When expanding this field, it may look
+ * something like:
+ * export const investigationOptions = t.intersection([
+ * t.exact(
+ *   t.type({
+ *     fields: NonEmptyArray(NonEmptyString),
+ *   })
+ * ),
+ * t.exact(
+ *   t.partial({
+ *     overide: t.boolean,
+ *   })
+ * ),
+ * ]);
+ *
+ */
+export type InvestigationOptions = t.TypeOf<typeof InvestigationOptions>;
+export const InvestigationOptions = t.exact(
+  t.type({
+    fields: NonEmptyArray(NonEmptyString),
+  })
+);
```

Comment:
Thank you for adding the description, Yara!

Nit: `InvestigationOptions` could be a little bit too generic name and could "attract" unrelated parameters in the future, thus making it harder to implement the diff algo. I'd personally stick to `InvestigationFields` and call the fields array `field_names`. But not a strong opinion, I could be lacking some context.

Reply:
I can see that point. We do want to keep it to params related to these fields, not any arbitrary investigation options, so I went ahead and updated with your suggestions.